### ISSUE TITLE
Deprecated: Array and string offset access syntax with curly braces is deprecated: components/com_jce/editor/libraries/classes/extensions/filesystem.php on line 116

### ISSUE THREAD

**Issue by ChristineWk** [widgetfactory/jce#96]
@ryandemmer 

Have a look please to this thread: https://www.phoca.cz/forum/viewtopic.php?f=35&p=167177#p167177
Thank you

information from user:
"Using php 7.3.29 things work ok but if I switch to
7.4.22 or 8.0.29 ..."

**Additional context**
Joomla 3.9.28
file: components/com_jce/editor/libraries/classes/extensions/filesystem.php on line 116

**Comment by ryandemmer** [comment]
The file in question does not contain curly braces offset in the current version - https://github.com/widgetfactory/jce/blob/master/components/com_jce/editor/libraries/classes/extensions/filesystem.php#L116

**Comment by ChristineWk** [comment]
Hi Ryan:

Thank you for your quick answer! Informed user about. 

What does this mean?:
/com_jce/editor/libraries/classes/browser.php on line 457
/com_jce/editor/libraries/classes/browser.php on line 473

**Comment by ryandemmer** [comment]
Please ensure the user is using the latest available version of JCE before posting an issue or bug report.

https://www.joomlacontenteditor.net/downloads/editor

**Comment by ChristineWk** [comment]
OK. Thanks Ryan for your investigation.

**Comment by jrjr2u** [comment]
Since I am the user in this case I felt I should chime in. Upon checking for updates in Joomla! 3.9.28 It responded that everything was up to date and no updates were available. I was at JCE 2.6.36 so I wonder why the newer version did not get found by the updater? I did install 2.9.10 and it fixed the errors. Appreciate the help!

**Comment by ryandemmer** [comment]
JCE 2.6.36 is VERY old, and I'm pretty sure you would have had multiple notifications to update over the years since that version was released.

**Comment by jrjr2u** [comment]
No I haven't had Joomla! offer any updates recently for JCE. Can't say when the last JCE update was found but I checked 5 different sites that I have and none of them found an update for JCE. That was the first thing I did when I found the errors. This issue came to light after updating PHP to the newer version available. I am on the default update Joomla! channel on all of my sites. I do check my sites at least monthly if not more often as I do try to keep up to date. I did download 2.9.10 and manually updated all my sites today and they are working correctly now. I chimed in here as I thought you might like to know what happened and what I had found. If I can help in any way let me know. Thanks
